src/main/java/com/google/sps/servlets/UserDetailsServlet.java

```diff
@@ -0,0 +1,119 @@
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/user")
+public class UserDetailsServlet extends HttpServlet {
+
+  /**
+   * doGet receives the request and returns the user's data with the given id
+   **/
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws ServletException, IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity userEntity = null;
+    Long id = Long.parseLong(request.getParameter("id"));
+    try {
+      userEntity = getUserById(datastore, id);
+      User userRequested = createUserObject(userEntity);
+      ArrayList<Recipe> recipesList = userRecipesList(id, datastore);
+
+      request.setAttribute("user", userRequested);
+      request.setAttribute("recipesList", recipesList);
+    } catch (EntityNotFoundException e) {
+      request.setAttribute("error", 1);
+    }
+    request.getRequestDispatcher("/user.jsp").forward(request, response);
+  }
+
+  private Entity getUserById(DatastoreService datastore, Long id)
+    throws IOException, EntityNotFoundException, NullPointerException {
+    Entity userEntity = datastore.get(KeyFactory.createKey("User", id));
+    return userEntity;
+  }
+
+  private User createUserObject(Entity userEntity) throws IOException {
+    User user = new User();
+    user.setId(userEntity.getKey().getId());
+    user.setName((String) userEntity.getProperty("name"));
+    user.setBio((String) userEntity.getProperty("bio"));
+    user.setImage((String) userEntity.getProperty("imgURL"));
+    user.setEmail((String) userEntity.getProperty("email"));
+
+    return user;
+  }
+
+  private ArrayList<Recipe> userRecipesList(
+    Long userId,
+    DatastoreService datastore
+  ) {
+    ArrayList<Recipe> recipesList = new ArrayList<>();
+    Filter propertyFilter = new FilterPredicate(
+      "author_id",
+      FilterOperator.EQUAL,
+      userId
+    );
+    Query q = new Query("Recipe").setFilter(propertyFilter);
+    PreparedQuery pq = datastore.prepare(q);
+
+    List<Entity> recipesEntityList = pq.asList(
+      FetchOptions.Builder.withDefaults()
+    );
+    for (Entity entity : recipesEntityList) {
+      recipesList.add(buildRecipe(entity));
+    }
+
+    return recipesList;
+  }
+
+  private Recipe buildRecipe(Entity recipeEntity) {
+    Long id = recipeEntity.getKey().getId();
+    String name = (String) recipeEntity.getProperty("title");
+    String imgURL = (String) recipeEntity.getProperty("imgURL");
+    String description =
+      "Lorem quam dolor dapibus ante, sit amet pellentesque turpis lacus eu ipsum. Duis quis mi ut tortor interdum efficitur quis at mi. Pellentesque quis mauris vel ligula commodo scelerisque. In vulputate quam nisl, vel sagittis ipsum molestie quis. Suspendisse quis ipsum a sem aliquam euismod mattis sed metus.";
```

Comment:
Move the 'lorem ipsum' to the database init as a description property placeholder. Here just retrieve the recipe description, so that this servlet works in the general case (when recipe have their own description) as well.